src/NLog/LayoutRenderers/CallSite and stacktraces/CallSiteLayoutRenderer.cs

```diff
@@ -130,12 +130,10 @@ StackTraceUsage IUsesStackTrace.StackTraceUsage
             {
 #if !SILVERLIGHT
                 if (FileName)
-                {
-                    return StackTraceUsage.Max;
-                }
+                    return SkipFrames == 0 ? StackTraceUsage.WithCallSiteSource : StackTraceUsage.WithSource;
+                else
 #endif
-
-                return StackTraceUsage.WithoutSource;
+                    return SkipFrames == 0 ? StackTraceUsage.WithCallSite : StackTraceUsage.WithoutSource;
```

Comment:
please create a helper for this

Reply:
FYI,

I'm trying to follow https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming) these days in NLog

Reply:
Done